guides/advanced_guides/filtering.md

```diff
@@ -0,0 +1,131 @@
+
+### Examples
+
+Suppose that you have a collection of movies, in the following JSON format:
+
+```json
+[
+	{
+		"id":"495925",
+		"title":"Doraemon the Movie: Nobita's Treasure Island",
+		"director": "Fujiko Fujio",
+		"poster":"https://image.tmdb.org/t/p/w1280/cmJ71gdZxCqkMUvGwWgSg3MK7pC.jpg",
+		"overview":"The story is based on Robert Louis Stevenson's Treasure Island novel."
+		,"release_date":1520035200
+	},
+	{
+		"id":"329996",
+		"title":"Dumbo",
+		"director": "Tim Burton",
+		"poster":"https://image.tmdb.org/t/p/w1280/279PwJAcelI4VuBtdzrZASqDPQr.jpg",
+		"overview":"A young elephant, whose oversized ears enable him to fly, helps
+		save a struggling circus, but when the circus plans a new venture, Dumbo
+		and his friends discover dark secrets beneath its shiny veneer.",
+		"release_date":1553644800
+	},
+	{
+		"id":"299536",
+		"title":"Avengers: Infinity War",
+		"director": "Joe Russo",
+		"poster":"https://image.tmdb.org/t/p/w1280/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
+		"overview":"As the Avengers and their allies have continued to protect the
+		world from threats too large for any one hero to handle...",
+		"release_date":1524618000
+	},
+	{
+		"id":"458723",
+		"title":"Us",
+		"director": "Jordan Peele",
+		"poster":"https://image.tmdb.org/t/p/w1280/ux2dU1jQ2ACIMShzB3yP93Udpzc.jpg",
+		"overview":"Husband and wife Gabe and Adelaide Wilson take their kids
+		to their beach house expecting to unplug and unwind with friends. But
+		as night descends, their serenity turns to tension and chaos when some
+		shocking visitors arrive uninvited.",
+		"release_date":1552521600
+	},
+		...
+]
+```
+
+You want to only show to your user movies that were released after a certain date, then you can use the following filter:
+
+```rust
+release_date > 795484800 // march 18, 1995
+```
+
+Now imagine that we want only the movies released after the 18 of march 1995, and directed by either Jordan Peel or Tim Burton, then you would use this filter:
+
+```rust
+release_date > 795484800 AND (director = "Tim Burton" OR director = "Jordan Peel")
+```
+
+Note that filtering on string is case insensitive. Here, the parentheses are mandatory, as `AND` has a higher precedence.
+
+Finally, imagine that you want to filter on "id". You would probably do this:
+
+```rust
+id = "299536"
+```
+
+However, this would be a very bad idea. Id uniquely identifies a movie, therefore, no matter how many candidates there are to the user query, only one will ever match. The filtering engine will have to go through all the candidate documents to find the only match. This is highly inefficient and should be avoided. We are working on a [faceted search](https://en.wikipedia.org/wiki/Faceted_search) feature, especialy optimized for this kind of need. It is important to understand that filtering and querying don't have the same performances.
```

Comment:
```suggestion
However, this would be a **very bad idea**. `Id` uniquely identifies a movie, therefore, no matter how many candidates there are to the user query, only one will ever match. The filtering engine will have to go through all the candidate documents to find the only match. This is highly inefficient and should be avoided. We are working on a [faceted search](https://en.wikipedia.org/wiki/Faceted_search) feature, especialy optimized for this kind of need. It is important to understand that filtering and querying don't have the same performances.
```